addrs.go

```diff
@@ -22,10 +22,11 @@ func (addr *Addr) Network() string {
 }
 
 // NewAddr creates a new Addr using the given host string
-func NewAddr(host string) *Addr {
+func NewAddr(scheme string, host string) *Addr {
```

Comment:
I'd prefer to deprecate this function and add a new one that accepts the scheme and a flag indicating whether this is IPv4 or IPv6. Then we can address your TODO in line 73 cleanly.

Reply:
Done. We don't need to differentiate between IPv4 and IPv6 anymore if we can just use `/dns`. Not sure what to name this new function. Currently it's:

```go
// NewAddrWithScheme creates a new Addr using the given host string. isSecure
// should be true for WSS connections and false for WS.
func NewAddrWithScheme(host string, isSecure bool) *Addr
```